File: javascript/rna-transcription/rna-transcription.js
var DnaTranscriber = function() { };

const D2R = {
  'G': 'C',
  'C': 'G',
  'T': 'A',
  'A': 'U'
};

DnaTranscriber.prototype.toRna = function(dna) {
  var rna = '';
  for (var c of dna) {
    rna += D2R[c];
  }
  return rna;
};

module.exports = DnaTranscriber;

File: erlang/rna-transcription/rna_transcription.erl
-module(rna_transcription).

-export([
         to_rna/1
        ]).

to_rna("G") ->
  "C";
to_rna("C") ->
  "G";
to_rna("T") ->
  "A";
to_rna("A") ->
  "U";
to_rna([R|Tail]) ->
  to_rna([R]) ++ to_rna(Tail);
to_rna([]) ->
  "".
